channel/channel.ml

```diff
@@ -195,12 +213,15 @@ module Make(Flow:V1_LWT.FLOW) = struct
     queue_obuf t;
     let l = List.rev t.obufq in
     t.obufq <- [];
-    Flow.writev t.flow l
-    >>= fun _ -> return_unit
+    Flow.writev t.flow l >>= function
+    | `Ok () -> Lwt.return_unit
+    | `Error (e : Flow.error) -> fail (Write_error (Flow.error_message e))
+    | `Eof -> fail (End_of_file)
 
   let close t =
-    flush t
-    >>= fun () ->
-    Flow.close t.flow
+    try_lwt
```

Comment:
You could use
```ocaml
Lwt.finalize (fun () -> flush t) (fun () -> Flow.close t.flow)
```
to avoid adding the `lwt.syntax` dependency (since `camlp4` is on its way out).